src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -895,13 +899,63 @@ public boolean isComplete() {
         first.setNewHead(head);
         heads.clear();
         heads.put(first.getId(),first);
+    }
 
-        // clean up heap
+    void cleanUpHeap() {
```

Comment:
`SerializableClassRegistry.release` was getting called on the right loader, but too soon—_before_ the final call to `saveProgram`, which would up adding the loader back to the registry. Not sure when this regressed—presumably at some point after the test was disabled.